travel-event-host-app/src/app/api/user/[id]/route.ts

```diff
@@ -0,0 +1,21 @@
+import { NextResponse } from 'next/server';
+import { connectMongoDB } from '@/lib/mongodb';
+import mongoose from 'mongoose';
+import User from '@/schemas/user';
+
+export async function GET(req: Request, { params }: any) {
+  //to get query use this, here you get query1 for example
+  const { searchParams } = new URL(req.url);
+  const query1 = searchParams.get('query1');
+  await connectMongoDB();
+
+  const id = params.id;
+  const isValidObjectId = mongoose.Types.ObjectId.isValid(id);
+  if (!isValidObjectId) {
+    return NextResponse.json({ message: 'Invalid ObjectId format' }, { status: 400 });
+  }
+
+  const userFound = await User.findById(id).select('-password');
+  if (userFound) return NextResponse.json({ userFound }, { status: 200 });
```

Comment:
See my comment about `SecureUser` below. Here you could create a SecureUser object and send that back in the response, instead of the document that we get from the `findById`

Reply:
OK I just looked at the documentation for .select. We should also exclude email and admin?

Reply:
no, we need to know the email and know if the user is a admin

Reply:
If the user is accessing their own ID then that sensitive information is ok to send back.
In the scenario when we are fetching all of the users that have registered for an event, the requestor doesn't need to know if all of those users are `admin`, their `password` or their `email` address. Does this make sense?

Reply:
yeah you are right